### PR TITLE
Catalyze Reactions

### PR DESCRIPTION
**Pull request by Charlie-Zheng**

### CODE REVIEW

File: pkg/enemy/damage.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func (t *Enemy) calc(atk *combat.AttackEvent, evt glog.Event) (float64, bool) {
 	emBonus := (2.78 * em) / (1400 + em)
 	var reactBonus float64
 	//check melt/vape
-	if atk.Info.Amped {
+	if atk.Info.AmpType == combat.Melt || atk.Info.AmpType == combat.Vaporize {
```
Comment (srliao) [inline pkg/enemy/damage.go:91]: Should add another flag here. Don't use amp flag

File: pkg/reactable/catalyze.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	a.Info.Amped = true
+	a.Info.AmpType = combat.Aggravate
+}
+
+func (r *Reactable) trySpread(a *combat.AttackEvent) {
+	if a.Info.Durability < ZeroDur {
+		return
+	}
+
+	switch a.Info.Element {
```
Comment (srliao) [inline pkg/reactable/catalyze.go:40]: The other ones are not done to figure out the trigger is correct, but to figure out which element is incoming to apply the correct calcs. The default case is technically unreachable.

So in this case the switch on `a.Info.Element` is redundant and should be removed. However, you still need to check for `r.Durability[attributes.Quicken]`

File: pkg/enemy/damage.go (hunk 1 above)
Comment (Charlie-Zheng) [reply pkg/enemy/damage.go:91]: What should I called the flag? Catalyzed?

**Comment by Charlie-Zheng** [comment]
I've updated the code to reflect the requested changes

**Pull request merged**